### ISSUE TITLE
Release 2.7

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#4734]
Create 2.7 release 

- [x] slack post
- [x] agree in team call (late Feb 2021)
- [x] Create branch for 2.7
- [x] Reassign any issues not being worked on to the next release
- [x] Update master to **2.8-SNAPSHOT** 
- [x] Update ref to presentation server image in chart(s)
- [x] Ensure any remaining fixes are merged 
- [x] Update branch version from 2.7-SNAPSHOT to 2.7
- [x] Remove release notes from branch 
- [x] update release notes & rename 
- [x] Verify notebooks (see #4842)
- [x] Check polymer UI 
- [x] Check node (ps) UI (see #4843)
- [x] Run CTS 
- [x] Run FVTs 
- [x] Check samples 
- [x] Check swagger doc 
- [x] Verify docker image/charts/compose 
- [x] create release candidate (manual github action)
- [x] Verify RC packaging on MC
- [x] Fix up maven artifacts as needed
- [x] Create final github releases (launches pipeline)
- [x] Verify artifacts all on maven central 
- [x] Release on maven central
- [x] ensure any fixes ported to master

**Comment by planetf1** [comment]
Note - this is the first release that will aim to use the new release pipeline - https://github.com/odpi/egeria/pull/4733
There may be a few teething issues & it's very likely any errant maven artifacts failing validation will need to be
 - fixed (if fixable ie containing code)
 - removed (if empty/no code)
 - OR publishing of artifact disabled - see #4464

**Comment by planetf1** [comment]
master will be branched for 2.7 on Sun 27 Feb

**Comment by planetf1** [comment]
Testing new release pipeline prior to continuing with testing ...

**Comment by planetf1** [comment]
Graph CTS results:
```
Date of conformance test:          2021-03-02T12:09:48.143+00:00
Number of tests that were run:     4965
Number of tests that failed:       0
Number of tests that were skipped: 0


id | name | description | profilePriority | conformanceStatus
-- | -- | -- | -- | --
0 | Metadata sharing | The technology under test is able to share met... | MANDATORY_PROFILE | CONFORMANT_FULL_SUPPORT
1 | Reference copies | The technology under test is able to save, loc... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
2 | Metadata maintenance | The technology under test supports requests to... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
3 | Dynamic types | The technology under test supports changes to ... | OPTIONAL_PROFILE | UNKNOWN_STATUS
4 | Graph queries | The technology under test supports graph-like ... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
5 | Historical search | The technology under test supports search for ... | OPTIONAL_PROFILE | CONFORMANT_NO_SUPPORT
6 | Entity proxies | The technology under test is able to store stu... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
7 | Soft-delete and restore | The technology under test allows an instance t... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
8 | Undo an update | The technology under test is able to restore a... | OPTIONAL_PROFILE | CONFORMANT_NO_SUPPORT
9 | Reidentify instance | The technology under test supports the command... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
10 | Retype instance | The technology under test supports the command... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
11 | Rehome instance | The technology under test supports the command... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
12 | Entity search | The technology under test supports the ability... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
13 | Relationship search | The technology under test supports the ability... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
14 | Entity advanced search | The technology under test supports the use of ... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
```

In-mem CTS results:

```
Date of conformance test:          2021-03-02T12:10:52.831+00:00
Number of tests that were run:     4965
Number of tests that failed:       0
Number of tests that were skipped: 0


id | name | description | profilePriority | conformanceStatus
-- | -- | -- | -- | --
0 | Metadata sharing | The technology under test is able to share met... | MANDATORY_PROFILE | CONFORMANT_FULL_SUPPORT
1 | Reference copies | The technology under test is able to save, loc... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
2 | Metadata maintenance | The technology under test supports requests to... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
3 | Dynamic types | The technology under test supports changes to ... | OPTIONAL_PROFILE | UNKNOWN_STATUS
4 | Graph queries | The technology under test supports graph-like ... | OPTIONAL_PROFILE | CONFORMANT_PARTIAL_SUPPORT
5 | Historical search | The technology under test supports search for ... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
6 | Entity proxies | The technology under test is able to store stu... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
7 | Soft-delete and restore | The technology under test allows an instance t... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
8 | Undo an update | The technology under test is able to restore a... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
9 | Reidentify instance | The technology under test supports the command... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
10 | Retype instance | The technology under test supports the command... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
11 | Rehome instance | The technology under test supports the command... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
12 | Entity search | The technology under test supports the ability... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
13 | Relationship search | The technology under test supports the ability... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT
14 | Entity advanced search | The technology under test supports the use of ... | OPTIONAL_PROFILE | CONFORMANT_FULL_SUPPORT

**Comment by planetf1** [comment]
Notebooks test ok, except for where they are marked as incomplete. JUst one minor issue in #4842

**Comment by planetf1** [comment]
Note: Polymer UI tested only in terms of basic UI comes up and login works

**Comment by planetf1** [comment]
@mandy-chessell I notice we don't have an updated implementation status map. Are you planning to update for 2.7? ie open-metadata-publication/website/roadmap/functional-organization-showing-implementation-status-for-2.7.png

@mandy-chessell @davidradl @yevgenmar @lcpopa @cmgrote @bogdan-sava @alexandra-bucur @grahamwallis @popa-raluca @marius-patrascu - you have all contributed to this release cc: @mstrelchuk @lpalashevski . 

Could you all consider if there is anything you wish to add to the release notes . 

I expect to send the code today so this needs to be actioned soon (though the content is in master and we link users back to it, so they'll still see updates). 

Please paste the content below, or create a new PR.

**Comment by cmgrote** [comment]
I would say most of what I've contributed is "groundwork" for more significant refactoring that is only fully available in the next release -- so I'd wait on mine...

One that I know I will depend on in 2.7 is @mandy-chessell 's changes to allow a `plugin-repository` repository to be configured through the admin services (a new back-end like Crux), as something distinct from a `repository-proxy` (a mapped existing metadata repository like IGC, Atlas, etc).

**Comment by planetf1** [comment]
Samples all ok within their current implementations.

**Comment by planetf1** [comment]
Known Issues:

* It is recommended to use a chromium-based browser such as Google Chrome or Microsoft Edge, or Apple Safari for the Egeria React UI. Some parts of the UI experience such as Dino currently experience problems with Firefox. See https://github.com/odpi/egeria-react-ui/issues/96 .
* When running the 'Understanding Platform Services' lab, ensure you run the 'egeria-service-config' notebook first and do not restart the python kernel before running this lab. See https://github.com/odpi/egeria/issues/4842 .

@mandy-chessell I presume you are going to aggregate and update

**Comment by planetf1** [comment]
Release 2.7 has now been released to maven central.

workflow was
1. Create a release in github (I usually copy the release notes for the text)
... actions kicks off automatically & As expected it took around 50 mins to build & push to maven central.
2. Login to oss.sonatype.org and navigate to staging repositories
3. Inspect the maven artifacts (I'd already tested before)
4. Click 'Close' to completion validating testing
... takes about 1 minute to go to closed, another 2 or so to allow the next action
5. Click 'Release' to release to the world
.. takes about 6-7 minutes to copy all files across

That's it. Compared to many days previously :-)
More could be automated, but it's better than before

I will hold off from posting to slack until we have the release notes updated

**Comment by planetf1** [comment]
Also worth noting that the artifacts are on maven central, but they take a while (hours-day) to be available via the front-facing index. This is as expected.

**Comment by grahamwallis** [comment]
Small release note on the graph repository performance improvements:

Release 2.7 includes performance improvements to the graph repository's search methods, which select an efficient query strategy based on the properties and type filters supplied to the search

Is that enough?

**Comment by planetf1** [comment]
@grahamwallis unless you want to add any clarification as to the impact - though that's always tricky with performance. So I think it's ok

**Comment by grahamwallis** [comment]
@planetf1 - agreed - performance figures would need to be very specific and we'd need to qualify them with machine specs etc etc. I don't think they would be all that interesting and wouldn't stay current for very long; also we don't publish performance figures for other aspects of the project. Suffice it to say that the graph queries that could benefit from the query changes got "much faster" :-)

**Comment by planetf1** [comment]
I have incorporated the comments so far - thanks.

**Comment by planetf1** [comment]
Releases have shipped and release notes have all the content available at the time
Release notes are in master so if anyone has additional updates feel free to update.
Closing.
